src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -111,6 +111,30 @@ internal static void GetUnderlyingTypeChain(NamedTypeSymbol underlyingTupleType,
             }
         }
 
+        /// <summary>
+        /// Gets flattened type arguments of the underlying type
+        /// which correspond to the types of the tuple elements left-to-right
+        /// </summary>
+        internal static void GetElementTypes(NamedTypeSymbol underlyingTupleType, ArrayBuilder<TypeSymbol> tupleElementTypes)
```

Comment:
Do we know that underlying type does not have some extra fields, always returns them in the right order, none is bad or missing?
I think as long as type arguments are tied to the type of the elements operating with arities and type arguments is more robust.